### PR TITLE
Add scrollo cache

### PR DESCRIPTION
**Pull request by eiais**
scrollo cache for performance and so you can see what scrollos got set after a stream

### CODE REVIEW

File: twitch.go [hunk 1]
```diff
@@ -180,13 +183,25 @@ func handleWebhook(w http.ResponseWriter, r *http.Request) {
 			cmd.Env = append(cmd.Env, "AUDIODEV=hw:1,0")
 			go cmd.Run()
 		} else if reward == "scrollo" {
-			f, err := os.Create("scrollo.txt")
+			scrolloFile := filepath.Join(".scrollocache/", params)
```
Comment (jqln-0) [inline twitch.go]: love to spend my simpbucks to put my favourite uplifting message on the screen; “../keys”

brings a tear to my eye every time it does

**Review by jqln-0 — CHANGES_REQUESTED** [review]

File: twitch.go (hunk 1 above)
Comment (eiais) [reply twitch.go]: hashing it to fix this, not sure how i could have missed it

**Review by jqln-0 — CHANGES_REQUESTED** [review]
This new logic looks pretty complicated! Would you mind adding some tests?

**Comment by eiais** [comment]
will add tests in a follow up PR

File: vendor/golang.org/x/sync/errgroup/errgroup.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	}
+	return g.err
+}
+
+// Go calls the given function in a new goroutine.
+//
+// The first call to return a non-nil error cancels the group; its error will be
+// returned by Wait.
+func (g *Group) Go(f func() error) {
+	g.wg.Add(1)
```
Comment (jqln-0) [inline vendor/golang.org/x/sync/errgroup/errgroup.go:53]: ?

**Comment by teisenbe** [comment]
Can you provide a benchmark showing the performance improvement?

File: twitch.go [hunk 3]
```diff
@@ -122,7 +123,11 @@ func handleWebhook(w http.ResponseWriter, r *http.Request) {
 	hmacKeys := [][]byte{[]byte(os.Getenv("dom_secret")), []byte(os.Getenv("sub_secret"))}
 	if !verifyWebhook(r, requestBody, hmacKeys) {
 		log.Println("failed to verify signature")
-		w.Write([]byte("you're my good puppy\n"))
+		_, err := w.Write([]byte("you're my good puppy\n"))
+		if err != nil {
+			log.Printf("failed to call kathryn a good puppy: %s", err)
```
Comment (teisenbe) [inline twitch.go:128]: A bit narcissistic, don't you think?

File: vendor/golang.org/x/sync/errgroup/errgroup.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	os.Link("keys", "scrollo.txt")
```
Comment (teisenbe) [inline vendor/golang.org/x/sync/errgroup/errgroup.go:54]: Huh, that's a weird thing in this library

**Pull request closed without merging**